src/components/Signup/SignupView.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { commonFetchJSON } from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Alert,
+  Bullseye,
+  Card,
+  CardBody,
+  CardFooter,
+  CardTitle,
+  PageSection,
+  Spinner,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons';
+import { Formik } from 'formik';
+import { UserSignupStatus } from '../../hooks/useSignupStatus';
+import ExternalLink from '../../shared/components/links/ExternalLink';
+import { SignupValues } from '../../types/signup';
+import SignupBanner from './SignupBanner';
+import SignupForm from './SignupForm';
+
+export type SignupViewProps = {
+  status: string;
+  loaded: boolean;
+  onStatusChange: React.Dispatch<React.SetStateAction<string>>;
+};
+
+const SignupView: React.FC<SignupViewProps> = ({
+  status = UserSignupStatus.NOT_SIGNEDUP,
+  loaded = true,
+  onStatusChange,
+}) => {
+  const initialValues: SignupValues = {
+    signUpText: '',
+  };
+
+  const handleSubmit = () => {
+    commonFetchJSON
+      .post('/api/v1/signup', {})
+      .then((res: any) => {
+        if (res.status === 202) onStatusChange(UserSignupStatus.PENDING_APPROVAL);
+      })
+      .catch();
+  };
+
+  if (!loaded) {
+    return (
+      <Bullseye>
+        <Spinner />
+      </Bullseye>
+    );
+  }
```

Comment:
Would it be a good idea to handle this in NamespacedPage itself? The check won't have to be repeated again that way. And `loaded` prop will also not be needed.